### PR TITLE
mx_path_canon() and change_folder_string() take a Buffer

### PR DESCRIPTION
**Pull request by roccoblues**
This makes changing `MxOps.path_canon()` (https://github.com/neomutt/neomutt/discussions/3902) easier.

### CODE REVIEW

File: mx.c [hunk 1]
```diff
@@ -1671,19 +1674,21 @@ struct Mailbox *mx_mbox_find2(const char *path)
...
+  mx_path_canon(buf, c_folder, NULL);
 
   struct Account *np = NULL;
   TAILQ_FOREACH(np, &NeoMutt->accounts, entries)
   {
-    struct Mailbox *m = mx_mbox_find(np, buf);
-    if (m)
+    struct Mailbox *m = mx_mbox_find(np, buf_string(buf));
+    if (m) {
+      buf_pool_release(&buf);
```
Comment (roccoblues) [inline mx.c:1686]: can I use `buf_pool_release()` with a Buffer I didn't get with `buf_pool_get()`? Or should I use `buf_free()` here and in line 1691?

Comment (flatcap) [reply mx.c:1686]: No, you'll need to `buf_free()` them.

The Pool is only good for short-term Buffers.
It keeps a cache and recycles Buffers on `buf_pool_release()` -- it normalises their size and zeros them.

File: index/dlg_index.c [hunk 2]
```diff
@@ -752,7 +752,6 @@ struct Mailbox *change_folder_notmuch(struct Menu *menu, char *buf, int buflen,
  * change_folder_string - Change to a different Mailbox by string
  * @param menu         Current Menu
  * @param buf          Folder to change to
- * @param buflen       Length of buffer
```
Comment (flatcap) [inline index/dlg_index.c:755]: Fix doxygen to match params

File: index/dlg_index.c [hunk 3]
```diff
@@ -784,7 +783,7 @@ void change_folder_string(struct Menu *menu, struct Buffer *buf, int *oldcount,
     }
     else
     {
-      mutt_error(_("%s is not a mailbox"), buf);
+      mutt_error(_("%s is not a mailbox"), buf_string(buf));
```
Comment (flatcap) [inline index/dlg_index.c:786]: pass a string to `%s`

File: mx.c [hunk 4]
```diff
@@ -38,9 +38,7 @@
 #include <sys/stat.h>
 #include <time.h>
 #include <unistd.h>
-#include "mutt/buffer.h"
 #include "mutt/lib.h"
-#include "mutt/pool.h"
```
Comment (flatcap) [inline mx.c]: For each library, `X/lib.h` will include/export everything

File: mx.c [hunk 5]
```diff
@@ -1400,9 +1398,9 @@ int mx_path_canon(struct Buffer *buf, const char *folder, enum MailboxType *type
   for (size_t i = 0; i < 3; i++)
   {
     /* Look for !! ! - < > or ^ followed by / or NUL */
-    if ((buf_at(buf,0) == '!') && (buf_at(buf,1) == '!'))
+    if ((buf_at(buf, 0) == '!') && (buf_at(buf, 1) == '!'))
```
Comment (flatcap) [inline mx.c:1401]: [clang-format](https://neomutt.org/dev/analysis/clang-format)
same for everything below

**Review by flatcap — APPROVED** [review]

**Pull request merged**